src/expressions/splitExpression.ts

```diff
@@ -200,7 +201,11 @@ export class SplitExpression extends ChainableExpression implements Aggregate {
 
   public getSelectSQL(dialect: SQLDialect): string[] {
     return this.mapSplits((name, expression) => {
-      if (['IP', 'SET/IP'].includes(expression.type)) {
+      if (
+        expression instanceof SqlRefExpression &&
+        ['IP', 'SET/IP'].includes(expression.type) &&
+        !expression.sql.match(/ip_search\(|ip_match\(/gi)?.length
```

Comment:
This works, but I think this would be more robust if it used the SQL parser instead of regex. I tested this out locally and it works:

```ts
function containsSqlFunction(expression: SqlRefExpression, functionNames: string[]): boolean {
  return (
    expression.parsedSql.type === 'function' &&
    functionNames.includes((expression.parsedSql as SqlFunction).functionName.toLowerCase())
  );
}

// ...
      if (
        expression instanceof SqlRefExpression &&
        ['IP', 'SET/IP'].includes(expression.type) &&
        !containsSqlFunction(expression, ['ip_search', 'ip_match'])
      ) {
// ...
```

Reply:
I like that better too. Good call out.